File: backend/app/models/request.py
from enum import Enum
from datetime import datetime

from sqlalchemy import DateTime, Enum as SQLEnum, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class RequestStatus(str, Enum):
    NEW = "new"
    IN_PROGRESS = "in_progress"
    DONE = "done"


class Request(Base):
    __tablename__ = "requests"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    contact: Mapped[str] = mapped_column(String(255), nullable=False)
    project_type: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)

    status: Mapped[RequestStatus] = mapped_column(
        SQLEnum(RequestStatus, name="request_status"),
        nullable=False,
        default=RequestStatus.NEW,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        server_default=func.now(),
    )

File: frontend/types/request.ts
export type RequestFormState = {
  name: string;
  contact: string;
  projectType: string;
  description: string;
  budget: string;
};

export const initialRequestForm: RequestFormState = {
  name: "",
  contact: "",
  projectType: "Telegram Bot",
  description: "",
  budget: "",
};

export type RequestStatus = "new" | "in_progress" | "done";

export type RequestItem = {
  id: number;
  name: string;
  contact: string;
  project_type: string;
  description: string;
  budget: string | null;
  status: RequestStatus;
  created_at: string;
};